webapp/components/survey/ItemLabelFunctionSelector/ItemLabelFunctionSelector.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import * as NodeDef from '@core/survey/nodeDef'
+import { useI18n } from '@webapp/store/system'
+
+import ButtonGroup from '@webapp/components/form/buttonGroup'
+import { useDispatch, useSelector } from 'react-redux'
+import { SurveyFormActions, SurveyFormState } from '@webapp/store/ui'
+
+const labelTypesKeys = {
+  byName: 'byName',
+  byLabel: 'byLabel',
+}
+const labelTypes = ({ i18n }) => [
+  {
+    key: labelTypesKeys.byName,
+    label: i18n.t(`common.${labelTypesKeys.byName}`),
+  },
+  {
+    key: labelTypesKeys.byLabel,
+    label: i18n.t(`common.${labelTypesKeys.byLabel}`),
+  },
+]
+
+const ItemLabelFunctionSelector = () => {
+  const i18n = useI18n()
+  const dispatch = useDispatch()
+
+  const nodeDefLabelFunction = useSelector(SurveyFormState.nodeDefLabelFunction)
+
+  const toggleLabelFunction = () => {
+    dispatch(SurveyFormActions.toggleNodeDefLabelFunction())
```

Comment:
this is a generic component, it can be used also in hierarch. pass the onChange function to this component